File: examples/anthropic-managed-agents/javascript/src/template-builder.ts
import { Template, defaultBuildLogger } from "e2b";

import { DEFAULT_BUILD_TEMPLATE_NAME } from "./constants.js";
import { template } from "./template.js";

export async function buildTemplate(templateName = DEFAULT_BUILD_TEMPLATE_NAME) {
  await Template.build(template, templateName, {
    cpuCount: 1,
    memoryMB: 1024,
    onBuildLogs: defaultBuildLogger(),
  });
  return { name: templateName };
}

File: examples/anthropic-managed-agents/python/anthropic_managed_agents_e2b/template_builder.py
from __future__ import annotations

from e2b import Template, default_build_logger

from anthropic_managed_agents_e2b.template import template


def build_template(*, template_name: str):
    return Template.build(
        template,
        template_name,
        cpu_count=2,
        memory_mb=4096,
        on_build_logs=default_build_logger(),
    )
